wandb/run_manager.py

```diff
@@ -79,9 +79,11 @@ def _thread_body(self):
                 self._file.seek(where)
             else:
                 self._on_read_fn(data)
+        data = self._file.read()
+        if data:
+            self._on_read_fn(data)
```

Comment:
This was the main bug, when the thread was stopped we could lose data.

Reply:
Any tailer such as history, events, or output.log uses this so previously on shutdown it wouldn't read the remaining data.

Reply:
I was more trying to ask why there is remaining data. Unless you understand why there was data left over, I'm not sure adding one more `read()` call is necessarily going to be a real fix. Maybe the real problem is that something is writing to history (or whatever) after all the IO is supposed to be shut down, and that might still be an issue after this change. I can see how this would improve things though.